### ISSUE TITLE
Opening a video crashes with KeyError: u'videos'

### ISSUE THREAD

**Issue by cretudorin** [CastagnaIT/plugin.video.netflix#610]
#### Your Environment
- Netflix add-on version: 1.2.0
- Operating system version/name:  CoreElec 9.2.2
- Device model:  Tanix TX92

Used Operating system:
* [ ] Android
* [ ] iOS
* [x] Linux
* [ ] OSX
* [ ] Raspberry-Pi
* [ ] Windows

### Describe the bug
Sometimes it crashed when trying to open some videos

#### Steps to reproduce the behavior
in most cases it crashed on this anime series  
https://www.netflix.com/watch/81033447

#### Possible fix
resources/lib/kodi/infolabels.py:257

#### Original: 
```
raw_data = api.get_video_raw_data(videoids_to_request)
for videoid in videoids_to_request:
    infos = get_info(videoid, raw_data['videos'][videoid.value], raw_data, profile_language_code)[0]
```

#### Fixed
```
raw_data = api.get_video_raw_data(videoids_to_request)
    if raw_data:
        for videoid in videoids_to_request:
            infos = get_info(videoid, raw_data['videos'][videoid.value], raw_data, profile_language_code)[0]
```

### Debug log
```
ERROR: [plugin.video.netflix (2)] Traceback (most recent call last):
    File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/run_addon.py", line 207, in run
    route([part for part in g.PATH.split('/') if part])
    File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/run_addon.py", line 38, in lazy_login_wrapper
    return func(*args, **kwargs)
    File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/run_addon.py", line 62, in route
    play(videoid=pathitems[1:])
    File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/common/videoid.py", line 295, in wrapper
    return func(*args, **kwargs)
    File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/common/logging.py", line 143, in timing_wrapper
    return func(*args, **kwargs)
    File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/navigation/player.py", line 93, in play
    [videoid, videoid_next_episode] if videoid_next_episode else [videoid])
    File "/storage/.kodi/addons/plugin.video.netflix/resources/lib/kodi/infolabels.py", line 259, in get_info_from_netflix
    infos = get_info(videoid, raw_data['videos'][videoid.value], raw_data, profile_language_code)[0]
KeyError: u'videos'
```

**Comment by CastagnaIT** [comment]
_First need a good FULL debug log instruction in readme_

I am not sure about the proposed resolution, it is not normal that the API not return any data or wrong data, i have tried to play 5 episodes without problem, then there are others influences

need more info to understand the circumstances and root cause:

1) the videos are played from library or within addon list?
2) the upnext is enabled?
3) if upnext is enabled, the issue happen only after you press play next from upnext window?
4) how profiles you have? and each profile have same language? if not describe all language of profiles
5) what is the language of the main profile?
6) if you have played from inside an addon list (so not kodi library) what profile/language used?

**Comment by mackal** [comment]
This is what I was getting after the 401 errors, which you are working on it at least know about

**Comment by CastagnaIT** [comment]
this not helping..
from my tests, the error 401 should happen only after first access or profile switch,
but it might depend on how the addon is set up and where you start the videos

i have already done the PR merge where i have tried to mitigate the 401 error problem

in any case if error 401 happen on play method
the playback would be stopped with http 401, if this is your case before try install daily build

In this Issue post there is no full log to understand better what happen,
but here the Issue author has posted only an `KeyError`

**Comment by mackal** [comment]
I've gotten 401 errors leading to this backtrace (can't confirm that is the case here) from Kodi library as well as playing through the add-ons menus


I think I have a log saved, I'll see if I can get it when I get back to my desktop

**Comment by cretudorin** [comment]
@CastagnaIT 
The issue doesn't occur every time, probably some request fails.
The easiest way to reproduce for me is to restart the device, wait for the addon to load and start playing 

-  1 => from within the addon

- 2, 3 => upnext is not installed

- 4, 5, 6 =>  2 profiles both have the language set to English,  everything it's set to English

log [here](https://gist.github.com/cretudorin/43a5a1b18c88fa35785ed3d611f7cf1a#file-kodi-log-L3726), relevant stuff starting with line 3726 (purged some mysql library updates)

I'm not sure if it's relevant, the device it's part of a VPN using zerotier

**Comment by CastagnaIT** [comment]
yes VPN is relevant with netflix can lead to several communication problems
I can analyze your case only by providing data with VPN disabled

**Comment by mackal** [comment]
> yes VPN or PROXY is relevant with netflix can lead to several communication problems
> I can analyze your case only by providing data with VPN disabled

I get it without VPN. Although, looking at their log, I don't see a 401 error before it happened like I was (could have been unrelated I guess)

**Comment by cretudorin** [comment]
It's not used to route traffic through vpn, it's just loading media sources with vpn and samba

**Comment by CastagnaIT** [comment]
@cretudorin well, but i will need of a full log with netflix debug info enabled
@mackal yes your problem is another case

**Comment by cretudorin** [comment]
updated the [log](https://gist.github.com/cretudorin/43a5a1b18c88fa35785ed3d611f7cf1a#file-kodi-log-L1709) to include debug

**Comment by CastagnaIT** [comment]
i have seen, and i think fired the problem in the request
try this and let me know
[plugin.video.netflix_1.2.0_20200501_t1.zip](https://github.com/CastagnaIT/plugin.video.netflix/files/4562906/plugin.video.netflix_1.2.0_20200501_t1.zip)

**Comment by cretudorin** [comment]
@CastagnaIT I've used provided archive and the crash seams to be gone.

**Comment by CastagnaIT** [comment]
excellent,
I really appreciate the support you sent me
thanks

**Comment by jakermx** [comment]
Just FYI
u means that the followed string should be formatted on UTF-8 basis
the Error you got points that it is not available on the region your where browsing from....i.e. I am from Mexico and some titles are not available here, I can add them to My List while using VPN but when I dont use it, I can see it listed but not available to play...
It should be great to have a human based rspond, saying..."mmmhh it is not available at the time" instead of getting an error

;)

**Comment by CastagnaIT** [comment]
VPN topics are not welcome here:
https://github.com/CastagnaIT/plugin.video.netflix/wiki/VPN-and-Proxy
